py/kubeflow/testing/util.py

```diff
@@ -448,6 +448,48 @@ def wait_for_deployment(api_client,
     "Timeout waiting for deployment {0} in namespace {1}".format(
       name, namespace))
 
+
+def wait_for_ingress(api_client,
+                        namespace,
+                        name,
+                        timeout_minutes=2):
+  """Wait for ingress to be ready.
+
+  Args:
+    api_client: K8s api client to use.
+    namespace: The name space for the deployment.
+    name: The name of the ingress.
+    timeout_minutes: Timeout interval in minutes.
+
+  Returns:
+    ingress: The ingress object describing the ingress.
+
+  Raises:
+    TimeoutError: If timeout waiting for ingress to be ready.
+  """
+  # Wait for tiller to be ready
+  end_time = datetime.datetime.now() + datetime.timedelta(
+    minutes=timeout_minutes)
+
+  ext_client_apps = k8s_client.ExtensionsV1beta1Api(api_client)
+
+  while datetime.datetime.now() < end_time:
+    ingress = ext_client_apps.read_namespaced_ingress(name, namespace)
+    # ready_replicas could be None
+    if ingress.status.load_balancer.ingress[0].hostname:
```

Comment:
I am not sure if there's state like if ingress.status.load_balancer == false or len(if ingress.status.load_balancer.ingress) == 0. Better to double check and make sure the code here doesn't throw unexpected exception

Reply:
In `ExtensionsV1beta1Api ` client, there's nothing like:

```
ingress.status.load_balancer == false or len(if ingress.status.load_balancer.ingress) == 0
```

I'll check other client set

Reply:
Use `try + catch` to handle exception instead